### ISSUE TITLE
Upsampling to a odd number

### ISSUE THREAD

**Issue by diwakar-ravichandran** [tensorflow/tensorflow#31759]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Binary 
- TensorFlow version (use command below): 1.13
- Python version: 3.5.2
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1
- GPU model and memory: V100 32GB

Current Behaviour:
Raises a Value Error: ambiguous dimension hence affecting workflow

Expected:
Upsample the tensor to desired size using single side padding

Code to reproduce the issue:

a = tf.random.uniform(shape=[1,45,60,3], dtype=tf.dtypes.float32)
kernel_size = (45/2,60/2)
b = AveragePooling2D(pool_size=kernel_size, strides=kernel_size, padding='same')(a)
c = UpSampling2D(size=kernel_size, data_format='channels_last', interpolation='bilinear')(b)

**Comment by gadagashwini** [comment]
@drnightbaron, Looks like code is incomplete. please provide the complete code to replicate the issue. Thanks!

**Comment by drnightbaron** [comment]
import tensorflow as tf
from tensorflow.keras.layers import *
a = tf.random.uniform(shape=[1,45,60,3], dtype=tf.dtypes.float32)
kernel_size = (45/2,60/2)
b = AveragePooling2D(pool_size=kernel_size, strides=kernel_size, padding='valid')(a)
c = UpSampling2D(size=kernel_size, data_format='channels_last', interpolation='bilinear')(b)

Thank you for the prompt response. This  is the code I am using. I restarted the kernel and I don't get the earlier error but the tensor sizes after upsampling is coming to (1, 44, 60, 3) rather than (1, 45, 60, 3)

**Comment by gadagashwini** [comment]
I am able to replicate the issue with Tensorflow 1.13. Please see the colab [gist](https://colab.research.google.com/drive/1_b--wZipoCTQlbr-WZ3SSCI9kzVKLIjU) here. Thanks!

**Comment by drnightbaron** [comment]
This is exactly the error that I faced. Thanks!

**Comment by jvishnuvardhan** [comment]
@drnightbaron Just curious what kind of use case you have that requires large `pool_size` and `strides`. In your example, shape of `a` is `[1,45,60,3]`. When you apply `pooling` on `a` it downsamples  `a` and resulting shape of `b` is [1,2,2,3]. Then you want to upsample `b` to `[1,45,60,3]`. Is that right? Thanks!

**Comment by drnightbaron** [comment]
Well I was trying to implement a recent paper in tensorflow . An implementation for pytorch is there. But not for tensorflow. Since Adaptive Average pooling is not a inbuilt function, I was trying to manually create a workaround for that, and I ran into this snag 

https://github.com/CommissarMa/Context-Aware_Crowd_Counting-pytorch for more reference

**Comment by yhliang2018** [comment]
The issue is already fixed in TF 2.x. For the arg `kernel_size` in both `AveragePooling2D` and `UpSampling2D`, it needs to be integers though.
